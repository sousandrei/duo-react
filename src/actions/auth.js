import { history } from '../routers/Router'
// import { get } from 'axios'

// const ENDPOINT = process.env.ENDPOINT


export const loginAction = (user) => ({
	type: 'LOGIN',
	user
})

export const login = (obj = {
	matricula: '',
	password: ''
}) => {
	return dispatch => {
		return new Promise((resolve, reject) => {

			//todo request
			let user = { ...obj, nome: 'Test' }

			//error
			// return reject('erro no login')

			//sucess
			history.push('/dashboard')
			resolve()

			dispatch(loginAction(user))

		})
	}
}

export const logoutAction = () => ({
	type: 'LOGOUT'
})

export const logout = () => {
	return dispatch => {
		dispatch(logoutAction())
		history.push('/')
	}
}
