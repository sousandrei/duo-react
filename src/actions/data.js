import axios from 'axios'

const DASHBOARD = process.env.DASHBOARD

export const getDataAction = data => ({
	type: 'GET_DATA',
	data
})

export const getData = () => {
	return async dispatch => {
		try {
			let result = await axios.get(DASHBOARD)
			dispatch(getDataAction(result.data))
		} catch (err) {
			console.error('could not retrieve data')
		}
	}
}
