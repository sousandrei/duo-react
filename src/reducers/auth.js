export default (state = {}, action) => {
	switch (action.type) {
		case 'LOGIN':
			localStorage.setItem('sirin', JSON.stringify(action.user))
			return { user: action.user }
		case 'LOGOUT':
			localStorage.removeItem('sirin')
			return {}
		default:
			return state
	}
}	