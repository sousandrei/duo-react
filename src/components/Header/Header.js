import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { history } from '../../routers/Router'
import { logout } from '../../actions/auth'


export class Header extends React.Component {

	state = {
		selectedKeys: [history.location.pathname.slice(1)]
	}

	handleClick = ({ key }) => {
		history.push(`/${key}`)
	}

	render() {
		return (
			<header className='header'>
				<div className='header__logo-links'>
					<img src='/images/logo_duo_ph.svg' />
				</div>
				<div className='header__buttons'>
					<button
						className='button'
						onClick={this.props.logout}>
						<img src='/icons/ic_logout_dg.svg' />
					</button>
				</div>
			</header>
		)
	}
}

Header.propTypes = {
	logout: PropTypes.func
}

const mapDispatchToProps = dispatch => {
	return {
		logout: () => dispatch(logout())
	}
}

export default connect(undefined, mapDispatchToProps)(Header)
