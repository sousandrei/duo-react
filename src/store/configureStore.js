import thunk from 'redux-thunk'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'

import auth from '../reducers/auth'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

export default () => {
	const store = createStore(
		combineReducers({
			auth
		}),
		composeEnhancers(applyMiddleware(thunk))
	)

	return store
}
