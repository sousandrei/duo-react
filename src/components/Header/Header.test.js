import React from 'react'
import thunk from 'redux-thunk'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import { logoutAction } from '../../actions/auth'

import Header, { Header as HeaderShallow } from './Header'

const mockStore = configureStore([thunk])

test('should render correctly', () => {
	const wrapper = shallow(<HeaderShallow />)
	expect(wrapper).toMatchSnapshot()
})

test('should render correctly with store', () => {
	const store = mockStore()
	const wrapper = shallow(<Header store={store} />)
	expect(wrapper).toMatchSnapshot()
})

test('should dispatch logout', () => {
	const store = mockStore()
	const wrapper = mount(<Header store={store} />)

	store.subscribe(() => {
		expect(store.getActions()).toEqual([logoutAction()])
	})

	wrapper.find('button')
		.last()
		.simulate('click')

})

test('should change path on click', () => {
	const store = mockStore()
	const wrapper = mount(<Header store={store} />)

	wrapper.find('MenuItem')
		.first()
		.simulate('click')

	expect(location.href.split('/')[3]).toEqual('dashboard')
})
