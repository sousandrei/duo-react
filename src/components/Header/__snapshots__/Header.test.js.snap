// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<header
  className="header"
>
  <div
    className="header__logo-links"
  >
    <img
      src="/images/caixa_sirin_hor.svg"
    />
    <Menu
      className=""
      defaultOpenKeys={Array []}
      defaultSelectedKeys={Array []}
      focusable={true}
      inlineIndent={24}
      level={1}
      mode="horizontal"
      onClick={[Function]}
      onDeselect={[Function]}
      onOpenChange={[Function]}
      onSelect={[Function]}
      prefixCls="rc-menu"
      selectable={true}
      selectedKeys={
        Array [
          "",
        ]
      }
      style={Object {}}
      subMenuCloseDelay={0.1}
      subMenuOpenDelay={0.1}
      triggerSubMenuAction="hover"
      visible={true}
    >
      <MenuItem
        key="dashboard"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onSelect={[Function]}
      >
        <h3>
          HOME
        </h3>
      </MenuItem>
      <MenuItem
        key="deputados"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onSelect={[Function]}
      >
        <h3>
          DEPUTADOS
        </h3>
      </MenuItem>
    </Menu>
  </div>
  <div
    className="header__buttons"
  >
    <button
      className="button"
      onClick={null}
    >
      <img
        src="/icons/ic_eng_cinza.svg"
      />
    </button>
    <button
      className="button"
    >
      <img
        src="/icons/ic_perfil_cinza.svg"
      />
    </button>
  </div>
</header>
`;

exports[`should render correctly with store 1`] = `
<Header
  logout={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
